6
12
18
24
30
36
42
48
54
60
66
72
78
84
90
96


# W3A17
Cấp số nhân
Cho 4 số a, b, c, d. Hãy kiểm tra xem 4 số này có thể theo thứ tự tạo thành 1 cấp số nhân với công bội nguyên theo đúng
thứ tự a, b, c, d hay không?
Gợi ý : Tìm công bội (b / a) nếu b chia hết cho a, sau đó lấy b nhân công bội và so sánh vs c, c nhân công bội và so sánh vs
d. 


In [2]:
a, b, c, d = map(int,input().split())

if a!= 0 and b%a==0 : 
    q = b//a
    if c == b*q and d == c*q :
        print('Yes')
    else : print('No')
elif a==0 and b==0 and c==0 and d==0 : 
    print('yes')
else : print('No')

yes


# W3A16
[Rounding]
Nhập vào một số thực: in ra kết quả làm tròn lên ,tròn xuống, làm tròn tới số nguyên gần nhất (không được sử dụng hàm
sẵn có của pyton). Các kết quả cách nhau một dấu cách 


In [ ]:
a,b = float(input())


# W3A15
[Xếp loại học lực]
Nhập vào điểm trung bình của một học sinh và in ra học lực của học sinh đó. Xếp loại học lực theo quy tắc:
- >= 8.0: Giỏi
- >= 6.5: Khá
- >= 5.0: Trung bình
- < 5.0: Yếu 


In [21]:
a, b, c, d = map(float,input().split())
TB = (a+b+c+d)/4
if TB >= 8 : print(f'{TB:.2f} Gioi')
elif 6.5<=TB<=8.0 : print(f'{TB:.2f} Kha')
elif 5<=TB<=6.5 : print(f'{TB:.2f} Trung Binh')
else  : print(f'{TB} Yeu')

7.50 Kha


# W3A14
[Giải phương trình bậc nhất]
Nhập vào hai số thực a và b của phương trình ax + b = 0. Tìm và in ra nghiệm của phương trình, kết quả làm tròn đến số
thập phân thứ 2. In ra "Vô nghiệm" hoặc "Vô số nghiệm" trong trường hợp tương ứng. 

In [19]:
a, b = map(float,input().split())
if a ==0 :
    if b ==0 : 
        print('vo so nghiem')
    else :
        print('vô nghiệm')
else :
    x = -b/a 
    print(f'{x:.2f} là nghiệm của pt')

    

-1.12 là nghiệm của pt


# W3A13
[Tính tiền điện]
Nhập vào số kWh điện tiêu thụ. Tính tiền điện theo các bậc sau:
- 0-50 kWh: 1500đ/kWh
- 51-100 kWh: 2000đ/kWh
- 100 kWh: 3000đ/kWh 


In [1]:
sodien = int(input())
if 0<=sodien<=50 :
    tien = sodien*1500
    print(tien) 
elif 51<=sodien<=100 :
    tien = sodien*2000
    print(tien)
else : 
    tien = sodien*3000
    print(tien)

120000


# W3A12
[Kiểm tra năm nhuận]
Nhập vào một số nguyên dương n là số năm, kiểm tra xem năm đó có phải là năm nhuận hay không. Năm nhuận là năm
chia hết cho 4 nhưng không chia hết cho 100, hoặc chia hết cho 400. Nếu là năm nhuận in ra "Yes", nếu không in ra "No"

In [23]:
n = int(input())
if (n%4==0 and n%100 != 0) or n%400==0 : 
    print(f'{n} la nam nhuan')
else : print(f'{n} khong phai la nam nhuan')
    

1234 khong phai la nam nhuan


# W3A11
[Phân loại tam giác]
Nhập vào 3 số nguyên dương a,b,c là độ dài 3 cạnh của một tam giác. Kiểm tra 3 số đã nhập có tạo thành một tam giác hợp
lệ hay không. Nếu không in ra "Không phải tam giác". Nếu có, phân loại tam giác đó là "Tam giác đều", "Tam giác cân", hay
"Tam giác thường". 

In [ ]:
from math import *
a, b, c = map(int,input().split()) #trong đó c là cạnh huyền
if a+b>c : 
    if a ==b ==c : print('Tam giac deu')
    elif c == sqrt(a**2 + b**2) : print('Tam giac vuong') # Hoặc elif c**2 == a**2 + b**2 : print('Tam giac vuong')
    elif a==b !=c : print('Tam giac can')
    else : print('Tam giac thuong')
else : print('Khong la tam giac')

Tam giac vuong


# W3A10
[Largest number]
Nhập vào 4 số nguyên. In ra màn hình số nguyên lớn nhất trong 4 số.

In [15]:
a, b, c, d = map(int,input().split())
print(max(a,b,c,d))
print(min(a,b,c,d))

4
1


# W3A9
[Triangle]
Nhập vào 3 số nguyên dương a, b, c. Kiểm tra xem a, b, c có cấu thành độ dài cạnh của 1 tam giác được không.
Nếu a, b, c cấu tạo thành được một tam giác, in ra "Yes". Ngược lại in ra "No".

In [13]:
a, b, c = map(int, input().split()) #Trong đó c là cạnh huyền của tam giác
if a+b >c : print('Yes')
else : print('No')



Yes


# W3A8
[Longer string]
Nhập vào hai chuỗi a, b gồm các chữ cái thường. In ra "True" nếu chuỗi a dài hơn chuỗi b. Ngược lại in ra "False".

In [ ]:
# Cách 1
a, b = (input().split())
if len(a) > len(b) : 
    print('True')
else : 
    print('False')

False


In [1]:
# Cách 2
a , b = input().split()
print(len(a) > len(b))

False


# W3A7
[Check negative numbers]
Nhập vào hai số nguyên a, b. Nếu cả hai số a, b đều âm thì in ra "Yes", ngược lại in ra "No". 


In [9]:
a, b = map(int,input().split())
if a < 0 and b<0 : print('Yes')
elif  a==0 and b==0 : print('0') ## cho thêm 
else : print('No')

No


# W3A6
[Even and Odd]
Nhập vào một số nguyên dương x, bạn hãy kiểm tra xem x là số chẵn hay lẻ. Nếu x là số chẵn, in ra "Even". Nếu x
là số lẻ, in ra "Odd". 


In [10]:
x = int(input())
if x%2==0 : print(f'{x} is Even')
else : print(f'{x} is Odd')

ValueError: invalid literal for int() with base 10: 'kad 12'

# W3A3
Kiểm tra xem một số có phải là lũy thừa của 2 hay không
Sử dụng các toán tử bitwise để xác định xem một số có phải là lũy thừa của 2 hay không.
Gợi ý: Đối với lũy thừa của 2, n & (n-1) == 0 

In [ ]:

n = int(input())
a = n > 0 and n &  (n-1) == 0
print(a)


# W3A2
Hoán đổi hai số không sd biến tạm thời
Sử dụng phép toán XOR bitwise để hoán đổi giá trị của hai biến. 

# W3A1
 Viết một chương trình nhập vào một số nguyên n và in ra các chữ số đảo ngược nhau. 


In [ ]:
#VD n = abcd có 4 chữ số in ra dcba
n = int(input())
dvi = n%10
chuc = n//10%10
tram = n//100%10
nghin = n//1000
a = dvi*1000 + chuc*100 + tram*10 + nghin
print(f'{a} là số đảo ngược của {n}')

5432 là số đảo ngược của 2345


#1 Nhập vào 1 số nguyên dương (0....9) , đọc số tương ứng
#2 2345 : Hai nghìn ba trăm bốn năm 
#3  129890 : Một trăm hai mươi chín nghìn tám trăm chín mươi

In [ ]:
n = int(input())
if 0 <= n <= 9 :
    if n ==0 :
        s = 'so khong'
    elif n==1 :
        s ='so mot'
    elif n == 2 :
        s = 'so hai'
    #.......
    else : 
        s = 'so chin'
    print(s)
else :
    print(f'{n} toi chua doc duoc')

so mot


In [ ]:
n = int(input())
if 99<n<1000 :
    dv = n%10
    chuc = (n//10)%10
    tram = n//100
    a = dv*100 + chuc*10 + tram
    print(a)
else : print(f'{n} loi ')


10000 loi 


In [ ]:
n = int(input())
if n>0 :
    print(f'{n} la so duong')
elif n<0 :
    print(f'{n} la so am')
else :
    print(f'{n} la so khong')

0 la so khong


# Nhập vào 1 số nguyên ktra xem là số chẵn hay lẻ


In [ ]:
n = int(input())
if (n%4==0 and n%100 !=0) or n%400==0 :
    print(f'{n} la nam nhuan')
else :
    print(f'{n} khong la nam nhuan')

148640 la nam nhuan


In [ ]:
n = int(input('Nhap vao 1 so:')) ## print(f'{n:.2f}')
if n%2==0 :
    print('Even')
else :
    print('odd')

odd
